mautic/mautic
# Webhook send test fix for query parameters and validation fix

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | n
| Related user documentation PR URL | na
| Related developer documentation PR URL | na
| Issues addressed (#s or URLs) | #1789
| BC breaks? | n
| Deprecations? | n

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:

This PR fixes two bugs.

1. It fixes the issue in #1789 where a URL with query parameters would lose the parameters when using the send test button because the request was cleaned with `InputHelper::clean()` rather than `InputHelper::url()` which changed the encoding of `&`. 
2. Fixes validation on the URL - before it allowed to save blank which resulted in a SQL error. 

#### Steps to test this PR:
1. Create a new http://requestb.in/
2. Create a new webhook and use the requestbin URL along with `?foo=bar&bar=foo
3. Send a test payload
4. View the request in - it should show a URL with the entire query
5. Save the webhook without a URL - it should now fail validation

### As applicable
#### Steps to reproduce the bug:
1. Same steps as above only in the request bin, the `bar=foo` will be missing and if you save a webhook without a URL, it'll fail with a SQL error.
